modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java

```diff
@@ -378,6 +396,32 @@ public void testFreeMarkerNotificationWithCommerceOrder() throws Exception {
 				TestPropsValues.getUserId()));
 	}
 
+	@Test
+	public void testFreeMarkerNotificationWithCustomService() throws Exception {
+		String body = LocalizationUtil.updateLocalization(
+			LocalizedMapUtil.getLocalizedMap(
+				HashMapBuilder.put(
+					LanguageUtil.getLanguageId(LocaleUtil.US),
+					"Test Service : ${objectFieldBusinessType}"
+				).build()),
+			null, "Body", LanguageUtil.getLanguageId(LocaleUtil.US));
+
+		int queueEntries = getNotificationQueueEntries().size();
```

Comment:
@rijema rename it to `notificationQueueEntriesSize`

Reply:
I'm worried about evaluating by the queue size, once I'll have to duplicate the call of _getNotificationQueueEntries() to avoid chaining. Do you think we could change the assert to start evaluating if the body of the last message contains the freemarker code? WIth this, I would have to 1) Create a variable to persists the body 2) Create a variable to get the queue size 3) Create a variable to get the last entry of the queue.